packages/firebase_auth/lib/firebase_auth.dart

```diff
@@ -192,4 +223,24 @@ class FirebaseAuth {
     final FirebaseUser currentUser = new FirebaseUser._(data);
     return currentUser;
   }
+
+  Future<Null> _callHandler(MethodCall call) async {
+    switch (call.method) {
+      case "onAuthStateChanged":
+        _onAuthStageChangedHandler(call);
+        break;
+    }
+    return null;
+  }
+
+  void _onAuthStageChangedHandler(MethodCall call) {
+    final Map<String, dynamic> data = call.arguments;
+    final FirebaseUser currentUser =
+        data != null ? new FirebaseUser._(data) : null;
+    _authStateChangedController
+        .where((StreamController<FirebaseUser> ctrl) => !ctrl.isClosed)
```

Comment:
You shouldn't need to do a check here, only controllers that have listeners will be in the `List`.